### PR TITLE
Janitorial: Refactor Domains List component 

### PR DESCRIPTION
**Pull request by aidvu**
- ES6ify
- Use localize
- Use redux analytics actions

Test:
- Open Domains
- Change primary
- Click add domain
- go back from a domain edit

Make sure no errors in console and all working well.

### CODE REVIEW

File: client/my-sites/domains/domain-management/list/index.jsx [hunk 1]
```diff
@@ -357,18 +370,77 @@ export const List = React.createClass( {
 					isSelected={ index === this.state.primaryDomainIndex }
 					selectionIndex={ index }
 					busy={ this.state.settingPrimaryDomain && index === this.state.primaryDomainIndex }
-					busyMessage={ this.translate( 'Setting Primary Domain…', { context: 'Shows up when the primary' +
-						' domain is changing and the user is waiting' } ) }
+					busyMessage={ this.props.translate(
+						'Setting Primary Domain…',
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/list/index.jsx]: :disappointed: 21 existing translations will be lost with this change.

Comment (aidvu) [inline client/my-sites/domains/domain-management/list/index.jsx]: Not really!

File: client/my-sites/domains/domain-management/list/item-placeholder.jsx [hunk 2]
```diff
@@ -8,7 +8,7 @@ import React from 'react';
  */
 import CompactCard from 'components/card/compact';
 
-const ListItemPlaceholder = React.createClass( {
+class ListItemPlaceholder extends React.PureComponent {
 	render() {
 		return (
 			<CompactCard className="domain-management-list-item is-placeholder">
```
Comment (delputnam) [inline client/my-sites/domains/domain-management/list/item-placeholder.jsx]: The linter expects these classNames to be prefixed with `list__`

File: client/my-sites/domains/domain-management/list/item.jsx [hunk 3]
```diff
@@ -72,22 +72,22 @@ const ListItem = React.createClass( {
 			return null;
 		}
 		return <Gridicon className="card__link-indicator" icon="chevron-right" />
```
Comment (delputnam) [inline client/my-sites/domains/domain-management/list/item.jsx]: Missing `;` 😞

File: client/my-sites/domains/domain-management/list/item-placeholder.jsx (hunk 2 above)
Comment (aidvu) [inline client/my-sites/domains/domain-management/list/item-placeholder.jsx]: Yep, but these are a PITA to fix as we need to move all the CSS. That's why I skip those. :)

**Comment by aidvu** [comment]
> in client/lib/mixins/analytics/index.js: ${xxxx} should have spaces like this ${ xxxx }.

We're getting rid of this one slowly, so no need to touch.

File: client/my-sites/domains/domain-management/list/item.jsx (hunk 3 above)
Comment (aidvu) [inline client/my-sites/domains/domain-management/list/item.jsx]: :scream:

**Pull request merged**